lib/pixel-mapper.cc

```diff
@@ -250,11 +258,29 @@ class VerticalMapper : public PixelMapper {
                                   int *matrix_x, int *matrix_y) const {
     int panel_width  = matrix_width  / chain_;
     int panel_height = matrix_height / parallel_;
+
     *matrix_x = (x % panel_width) +  int(y/panel_height)* panel_width;
     *matrix_y = (y % panel_height) + int(x/panel_width) * panel_height;
+
+    if (z_) {
+	int x_panel_offset_cnt = *matrix_x / panel_width;
+	int x_panel_offset = x_panel_offset_cnt * panel_width;
+	int y_panel_offset_cnt = *matrix_y / panel_height;
+	int y_panel_offset = y_panel_offset_cnt * panel_height;
+
+	if (x_panel_offset_cnt % 2) {
+
+	    *matrix_x = panel_width - (*matrix_x - x_panel_offset) - 1;
```

Comment:
here, you're overwriting calculation we already did above; we should make more clear that we either do the one thing or the other (see other comment above).

Also why not have everything in one expression instead of fixing up the value below, which makes it much harder to follow.
```
  *matrix_x = panel_width - (*matrix_x - x_panel_offset) - 1 + x_panel_offset;
```

Reply:
I'm pretty sure the optimizer makes it all the same in the end (i.e. it get collapsed) and it was slightly easier to read, but I agree that this one is not a big deal to collapse, so I just did so.
Obviously you can also do +2 * x_panel_offset instead of  -(- x_panel_offset) + x_panel_offset  but I wager that would be much less obvious to read, and the compiler should definitely do it for us.